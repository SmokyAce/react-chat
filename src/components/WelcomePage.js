import React from 'react';
import { withStyles } from 'material-ui/styles';
import { Redirect } from 'react-router-dom';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';
import Tabs, { Tab } from 'material-ui/Tabs';
import LoginForm from './LoginForm';
import SignUpForm from './SignUpForm';

const styles = theme => ({
  content: {
    display: 'flex',
    paddingTop: '64px',
    justifyContent: 'center'
  },
  authForm: {
    marginTop: '24px',
    width: '500px'
  },
  tabWrapper: {
    padding: theme.spacing.unit * 3
  }
});

class WelcomePage extends React.Component {
  state = {
    activeTab: 0
  };

  handleTabChange = (event, value) => {
    this.setState({ activeTab: value });
  };

  render() {
    const { classes, login, signup, isAuthenticated } = this.props;
    const { activeTab } = this.state;

    if (isAuthenticated) {
      return <Redirect to='/chat' />;
    }

    return (
      <div>
        <AppBar color='primary'>
          <Toolbar>
            <Typography variant='title' color='inherit' noWrap>
              DogeCodes React Chat
            </Typography>
          </Toolbar>
        </AppBar>
        <div className={classes.content}>
          <Paper className={classes.authForm}>
            <AppBar position='static' color='default'>
              <Tabs
                value={activeTab}
                onChange={this.handleTabChange}
                indicatorColor='red'
                fullWidth
              >
                <Tab label='LOGIN' />
                <Tab label='SIGN UP' />
              </Tabs>
            </AppBar>
            <div className={classes.tabWrapper}>
              {activeTab === 0 && <LoginForm onSubmit={login} />}
              {activeTab === 1 && <SignUpForm onSubmit={signup} />}
            </div>
          </Paper>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(WelcomePage);
