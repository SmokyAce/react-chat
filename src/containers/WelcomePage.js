import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import WelcomePage from '../components/WelcomePage';
import { login, signup, receiveAuth } from '../actions';

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      login,
      signup,
      receiveAuth
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(WelcomePage);
