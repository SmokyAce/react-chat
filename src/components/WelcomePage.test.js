/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import WelcomePage from './WelcomePage';

jest.mock('./LoginForm', () => () => 'LoginForm');
jest.mock('./SignUpForm', () => () => 'SignupForm');
jest.mock('./ErrorMessage', () => () => './ErrorMessage');

const mockProps = {
  signup: jest.fn(),
  login: jest.fn(),
  receiveAuth: jest.fn(),
  isAuthenticated: false,
  error: null,
};

describe('<WelcomePage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter initialEntries={['/']}>
        <Route path="/(welcome)?" render={props => <WelcomePage {...mockProps} {...props} />} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  /* eslint-disable react/jsx-indent */
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/']}>
          <Route path="/(welcome)?" render={props => <WelcomePage {...mockProps} {...props} />} />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
