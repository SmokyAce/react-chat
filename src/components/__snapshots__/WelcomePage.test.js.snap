// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root-114 MuiPaper-elevation4-120 MuiAppBar-root-106 MuiAppBar-positionFixed-107 MuiAppBar-colorPrimary-112 mui-fixed"
  >
    <div
      className="MuiToolbar-root-141 MuiToolbar-gutters-142"
    >
      <h2
        className="MuiTypography-root-143 MuiTypography-title-149 MuiTypography-colorInherit-162 MuiTypography-noWrap-159"
      >
        DogeCodes React Chat
      </h2>
    </div>
  </header>
  <div
    className="WelcomePage-content-103"
  >
    <div
      className="MuiPaper-root-114 MuiPaper-elevation2-118 MuiPaper-rounded-115 WelcomePage-authForm-104"
    >
      <header
        className="MuiPaper-root-114 MuiPaper-elevation4-120 MuiAppBar-root-106 MuiAppBar-positionStatic-110 MuiAppBar-colorDefault-111"
      >
        <div
          className="MuiTabs-root-167"
        >
          <div
            className="MuiTabs-flexContainer-168"
          >
            <div
              className="MuiTabs-scroller-169 MuiTabs-fixed-170"
              onScroll={[Function]}
              role="tablist"
              style={
                Object {
                  "marginBottom": 0,
                }
              }
            >
              <div
                className="MuiTabs-flexContainer-168"
              >
                <button
                  aria-selected={true}
                  className="MuiButtonBase-root-192 MuiTab-root-176 MuiTab-textColorInherit-178 MuiTab-textColorInheritSelected-185 MuiTab-fullWidth-187"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-188"
                  >
                    <span
                      className="MuiTab-labelContainer-189"
                    >
                      <span
                        className="MuiTab-label-190"
                      >
                        LOGIN
                      </span>
                    </span>
                  </span>
                  <span
                    className="MuiTouchRipple-root-198"
                  />
                </button>
                <button
                  aria-selected={false}
                  className="MuiButtonBase-root-192 MuiTab-root-176 MuiTab-textColorInherit-178 MuiTab-fullWidth-187"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="tab"
                  style={undefined}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiTab-wrapper-188"
                  >
                    <span
                      className="MuiTab-labelContainer-189"
                    >
                      <span
                        className="MuiTab-label-190"
                      >
                        SIGN UP
                      </span>
                    </span>
                  </span>
                  <span
                    className="MuiTouchRipple-root-198"
                  />
                </button>
              </div>
              <span
                className="MuiTabIndicator-root-195 MuiTabs-indicator-175"
                style={
                  Object {
                    "backgroundColor": "red",
                    "left": 0,
                    "width": 0,
                  }
                }
              />
            </div>
          </div>
        </div>
      </header>
      <div
        className="WelcomePage-tabWrapper-105"
      >
        LoginForm
      </div>
    </div>
  </div>
  ./ErrorMessage
</div>
`;
